packages/core/utils/src/dml/helpers/create-mikro-orm-entity.ts

```diff
@@ -170,6 +174,39 @@ export function createMikrORMEntity() {
       return
     }
 
+    /**
+     * Defining an id property
+     */
+    if (field.dataType.name === "id") {
+      const IdDecorator = field.dataType.options?.primaryKey
+        ? PrimaryKey({
+            columnType: "text",
+            type: "string",
+            nullable: field.nullable,
+          })
+        : Property({
+            columnType: "text",
+            type: "string",
+            nullable: field.nullable,
+          })
+
+      IdDecorator(MikroORMEntity.prototype, field.fieldName)
+
+      /**
+       * Hook to generate entity within the code
+       */
+      MikroORMEntity.prototype.generateId = function () {
+        this.id = generateEntityId(this.id, "acc")
```

Comment:
@adrien2p What should be the prefix of id generation. Do we follow some convention?

Reply:
ahah I ve put a comment about it, we generally use a short name that correspond to the model name and that is meaningful a bit. so it should be configurable and if not provided we can infer it from the model name I would say. Something along the line: transform to snake case, split on `_`, if length 1 then the first 3 letters, if length 2 then the first two letters of each and if more then the first letter of each. wdyt?